test/e2e/scheduling/priorities.go

```diff
@@ -255,6 +258,10 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(tolePod.Spec.NodeName).To(Equal(nodeName))
 	})
+
+	It("should be able to read and write priorityClass", func() {
```

Comment:
It is a bit unfortunate that our node scoring functions are called "priority" functions. This file that you have modified test those priority functions and has nothing to do with pod priority and PriorityClasses.

Reply:
I understand @bsalamat, few operations for PriorityClasses viz POST, GET are covered in "preemption.go" but DeleteCollection, PATCH, PUT operations are not covered. My intention is to cover all [READ and Write](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#priorityclass-v1alpha1-scheduling-k8s-io) operations on PriorityClass along with few options (globalDefault) and **below scenarios** -
1. Should be able to create PC with int32(value)
2. **Only one** PriorityClass with `globalDefault=true` can exist in the system.
3. As default priorities for pod is zero, creating `globalDefault enabled` PC, should not change priorities of existing pods but would be applied for all pods created after global PC added.
(e.g PC Added with globalDefault=true && value=10 would lead to all pod creation with **default priority of 10** and existing pods to remain on priority 0 )
4. If PC is deleted, running pods would run with no change in priority and new pods can not be created with deleted PC name.
5. Patching PC with few updation like(value change, LabelSelector update etc) or PUT/REPLACE whole PC with new PC Object. 
6. Clean up all PriorityClasses and Pods. (to cover deleteCollection api endpoint of PCs)

Scenarious #1 would ensure, api-server is started with required flags(admission-control), alpha/beta features enabled or ability to create PriorityClasses.  

You can guide me on this to findout its effectiveness or few more points to cover. **Our main goal is to cover all API endpoints for PriorityClass in e2e tests and consider it as conformance test on later stage.**

Though PriorityClass is key component for Preemption, we can include this inside `preemption.go`. **What would be your thought on this?**

TODO for me could be-
1. List down all covered and uncovered PriorityClass operations in `preemption.go`
2. Better place to discuss such scenarios for various api-groups with respective working-groups, owners. ```(suggestions appreciated)```
3. Walkthrough on [scheduling design-proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling) for PriorityClass and Preemption.

cc @AishSundar @mithrav @timothysc

Reply:
> Though PriorityClass is key component for Preemption, we can include this inside preemption.go

IMO, e2e test cases in preemption should focus on core preemption functionality. E2E test cases related to operations on priority class objects should be in the pkg which is the main way of creating/deleting/modifying priority class objects and that is kubectl (or specifically sub commands of kubectl like create, get, delete, etc).

Reply:
@aveshagarwal, I'm primarily focusing on covering API endpoints under **conformance test suit to verify compatibility with all k8s providers**, covering important scenarios. (under conformance program). This is the reason I prefer to keep them in `test/e2e/scheduling`.

> E2E test cases related to operations on priority class objects should be in the pkg ...

pkg means `k8s.io/kubernetes/pkg/` ? With my understanding, this is meant for core functionality not e2e. `(correct me if I missunderstood)`
Though kubectl can cover all api-groups operations but here our primary focus is to **cover api-groups with conformance test** scenarios for PriorityClass.

Reply:
@bsalamat, What would be your suggestion on [listed scenarios](https://github.com/kubernetes/kubernetes/pull/62895#discussion_r184761911) ? Should we involve respective owners into discussion?

As our main goal is to cover every API groups funcationlity to be covered under conformance tests, you can suggest good approach to get suggestions/discussions/approval to speedup development process. Awaiting for Response..